org.eclipse.sisu.plexus/src/main/java/org/codehaus/plexus/component/configurator/converters/composite/ObjectWithFieldsConverter.java

```diff
@@ -72,6 +91,39 @@ public Object fromConfiguration( final ConverterLookup lookup, final PlexusConfi
         }
     }
 
+    private Object fromConstructorTakingString( final Class<?> type, String value, ComponentConfigurationException precedingException ) throws ComponentConfigurationException
+    {
+        try
+        {
+            try
+            {
+                Constructor<?> constructor = type.getConstructor( String.class );
+                return constructor.newInstance( value );
+            }
+            catch ( NoSuchMethodException e )
+            {
+                // suppress emitting this exception if there is a preceding exception
+                if ( precedingException != null )
+                {
+                    throw precedingException;
+                }
+                else
+                {
+                    throw e;
+                }
+            }
+        }
+        catch ( InstantiationException | IllegalAccessException | IllegalArgumentException | InvocationTargetException | NoSuchMethodException | SecurityException e )
```

Comment:
you could replace several of these with their common super-type, `ReflectiveOperationException` if you want